app/jobs/email_privacy_policy_update_job.rb

```diff
@@ -0,0 +1,43 @@
+class EmailPrivacyPolicyUpdateJob < ApplicationJob
+  def perform(petition)
+    terminating = false
+
+    worker = trap("TERM") do
+      terminating = true
+      worker.call
+    end
+
+    Appsignal.without_instrumentation do
+      petition.signatures.find_each do |signature|
+        create_privacy_notification!(signature)
+        send_email(signature)
+
+        if terminating
+          reschedule_job(petition)
+          return true
+        end
+      rescue ActiveRecord::RecordNotUnique
+        next
+      end
+    end
+  ensure
+    trap("TERM", worker)
+  end
+
+  private
+
+  def create_privacy_notification!(signature)
+    PrivacyNotification.create!(id: signature.uuid)
+  end
+
+  def send_email(signature)
+    PetitionMailer
+      .with(signature: signature)
+      .privacy_policy_update_email
+      .deliver_now
```

Comment:
The pattern in the original PR had this as creating a separate job called `DeliverDissolutionNotificationJob` - was there a reason for changing it? There's a performance issue to bear in mind when sending emails in a loop like this - each send will take approximately 500ms meaning the whole job for the largest petition (approx 700k signatures) will take nearly 4 days to run and be interrupted 4 or 5 times. By enqueuing the separate job it takes a few ms to insert the record per signature and it'll complete in a few hours. We can then scale up/down workers as needed to manage the queue size and daily limit of emails.

Also there's no handling of errors from the email sending - the separate job inherits from `EmailJob` which has a bunch of stuff for rescuing from network errors, etc. and retrying the job in the event of failure.

Reply:
Eep...that makes sense! It wasn't a deliberate decision - I think I was at least using `deliver_later` at some point and it got lost in a refactor ☹️ 

Amended now!